lib/segment/analytics/client.rb

```diff
@@ -355,7 +358,7 @@ def event attrs
       end
 
       def check_user_id! attrs
-        fail ArgumentError, 'Must supply either user_id or anonymous_id' unless attrs[:user_id].present? || attrs[:anonymous_id].present?
+        attrs[:anonymous_id] = (attrs[:user_id] ? Digest::MD5.hexdigest(attrs[:user_id].to_s) : SecureRandom.uuid) unless attrs[:anonymous_id]
```

Comment:
My Ruby is as good as my Japanese so please let me know if this is too unreadable or not idiomatic

Reply:
```ruby
attrs[:anonymous_id] ||= attrs[:user_id] ? Digest::MD5.hexdigest(attrs[:user_id].to_s) : SecureRandom.uuid 
```